nym-api/src/unstable_routes/account.rs

```diff
@@ -0,0 +1,207 @@
+            node_id: d.node_id,
+            proxy: d.proxy.clone(),
+        })
+        .collect();
+
+    // 3. get the current reward for each active delegation
+    // calculate rewards from nodes this delegator delegated to
+    let mut rewards_map: HashMap<&NodeId, NyxAccountDelegationRewardDetails> = HashMap::new();
+    for delegation in og_delegations.iter() {
+        let node_id = &delegation.node_id;
+
+        if let Some(nym_node_details) = nym_nodes.get(node_id) {
+            match nym_node_details
+                .rewarding_details
+                .determine_delegation_reward(delegation)
+            {
+                Ok(delegation_reward) => {
+                    rewards_map.insert(
+                        node_id,
+                        NyxAccountDelegationRewardDetails {
+                            node_id: delegation.node_id,
+                            rewards: decimal_to_coin(delegation_reward, "unym"),
+                            amount_staked: nym_node_details.original_pledge().clone(),
+                            node_still_fully_bonded: !nym_node_details.is_unbonding(),
+                        },
+                    );
+                }
+                Err(err) => {
+                    warn!(
+                        "Couldn't determine delegations for {} on node {}: {}",
+                        &account_id, node_id, err
+                    )
+                }
+            }
+        }
+    }
+
+    // 4. make the map of rewards into a vec and sum the rewards and delegations
+    let accumulated_rewards: Vec<NyxAccountDelegationRewardDetails> =
+        rewards_map.values().cloned().collect();
+
+    let mut claimable_rewards = 0u128;
+    let mut total_delegations = 0u128;
+    for r in &accumulated_rewards {
+        claimable_rewards += r.rewards.amount.u128();
+        total_delegations += r.amount_staked.amount.u128();
+        total_value += r.rewards.amount.u128();
+        total_value += r.amount_staked.amount.u128();
+    }
+
+    // 5. get vesting account details (if any): none because everyone has already fully vested
+    let vesting_account: Option<NymVestingAccount> = None;
+
+    if let Some(vesting_account) = vesting_account.clone() {
+        total_value += vesting_account.locked.amount.u128();
+        total_value += vesting_account.spendable.amount.u128();
+    }
+
+    // 6. get operator rewards: is this an operator?
+    let mut operator_rewards = 0;
+
+    for (_, node_details) in nym_nodes.iter() {
+        if address == node_details.bond_information.owner.as_str() {
+            let pending_operator_reward = node_details.pending_operator_reward().amount.u128();
+
+            // add operator rewards
+            operator_rewards += pending_operator_reward;
+
+            // add to totals
+            total_value += pending_operator_reward;
+        }
+    }
+
+    // 7. convert totals
+    let claimable_rewards = Coin::new(claimable_rewards, "unym");
```

Comment:
I don't really like the hardcoded `unym` everywhere. what if some testnet uses different denom? can you just pull it from one of many `Coin`s you got it at the start?